core/schema/modulesource.go

```diff
@@ -485,6 +485,23 @@ func (s *moduleSchema) moduleSourceDependencies(
 					return fmt.Errorf("failed to create module source from dependency: %w", err)
 				}
 
+				currentDep, err := depSrc.Self.Symbolic()
+				if err != nil {
+					return err
+				}
+
+				for _, dep := range src.Self.WithoutDependencies {
+					removedDep, err := dep.Self.Source.Self.Symbolic()
+					if err != nil {
+						return err
+					}
+
+					// the currentDep is being uninstalled
+					if currentDep == removedDep {
+						return nil
+					}
```

Comment:
Should we cover non-exact matches? e.g. if I install `github.com/foo/bar@v1.2.3`, and uninstall `github.com/foo/bar` should it match?

I think it probably should (unless there's another potential match, in which case we should error with a special "ambiguous dependency" - maybe tricky to do based on current code structure).

Reply:
right now it works like that. if we install `github.com/foo/bar@v1.2.3`, we can uninstall using exact url as well as without version.

however currently it does not work if we try to use just the name of the module as specified in `dagger.json`. do we need to support that as well?